devito/dse/rewriters.py

```diff
@@ -408,5 +409,26 @@ def _extract_sum_of_products(self, cluster, template, **kwargs):
 
 class CustomRewriter(AggressiveRewriter):
 
+    passes_mapper = {
+        'extract_sum_of_products': AggressiveRewriter._extract_sum_of_products,
+        'factorization': AggressiveRewriter._factorize,
+        'eliminate_inter_stencil_redundancies':
+            AggressiveRewriter._eliminate_inter_stencil_redundancies
+
+    }
+
+    def __init__(self, passes, template=None, profile=True):
+        try:
+            passes = passes.split(',')
+        except AttributeError:
+            # Already in tuple format
+            if not all(i in CustomRewriter.passes_mapper for i in passes):
+                raise DSEException("Unknown passes `%s`" % str(passes))
+        self.passes = passes
+        self.template = template
+        self.profile = profile
+        super(CustomRewriter, self).__init__()
```

Comment:
as I think I said in a previous message, this is not how you should invoke the superclass constructor. You should also pass the arguments that the superclass is expected to handle, such as `template` and `profile`

Reply:
`super(CustomRewriter, self).__init__(self.profile, self.template)` ?